src/common/configuration_store/configuration_store.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+#include "eeprom_access.hpp"
+#include "filament.hpp"
+#include <utility>
+#include <tuple>
+#include "configuration_store_structure.hpp"
+#include "item_updater.hpp"
+
+/**
+ * This class is configuration store which provides easy access to its values.
+ * It initializes the values from eeprom and persists them into it on change.
+ * Data is written to eeprom only if the value is non-default or if it is overwriting non-default data with default data
+ *
+ * To add new item to configuration store, you need to add it to json from which it is generated
+ *
+ * To access the values get singleton instance of the store and access the items like members of any other class. To change the items use their Set and Get methods.
+ */
+template <class CONFIG>
+class ConfigurationStore : public CONFIG {
```

Comment:
/suggestion/ The configuration store is implemented across several files and its public API consists of more classes etc. What about placing all of it into some namespace?